src/shared/compounders/AMMCompounderBase.sol

```diff
@@ -200,6 +216,8 @@ abstract contract AMMCompounderBase is ReentrancyGuard, ERC4626 {
     /// @return _assets - The amount of assets deposited.
     // slither-disable-next-line reentrancy-no-eth
     function mint(uint256 _shares, address _receiver) external override nonReentrant returns (uint256 _assets) {
+        if (_shares >= maxMint(msg.sender)) revert InsufficientDepositCap();
```

Comment:
Why do we need both, `maxMint` and `maxDeposit`?

Reply:
`mint` and `deposit` have different functionality, similar to `withdraw` and `redeem`.

- In the `mint`/`redeem` functions the user specifies how much _shares_ he wants to use.
- In the `deposit`/`withdraw` functions the user specifies how much _assets_ he wants to use.

Each function has it's own limit/preview view functions.